### DISCUSSION TITLE
Whistleblower Extension (onlyEditor vs onlyWhistleBlower)

### DISCUSSION THREAD

**Discussion by abhilashi** [semaphore-protocol/semaphore#108] [General]
https://github.com/semaphore-protocol/semaphore/blame/main/contracts/extensions/SemaphoreWhistleblowing.sol#L79

I'm wondering why only editors are allowed to publish a leak. Why not let any whistle blower publish a leak? @cedoor is there a dependency here?

**Comment by cedoor** [comment]
Hi @abhilashi, we could remove that modifier and allow everyone to publish a leak. However, the person who publishes should not be the whistle blower, because the transaction is public and their address would be revealed.
